### PR TITLE
Create thumbnails for django-hero logos

### PR DESCRIPTION
**Pull request by relekang**
Related to #364

### CODE REVIEW

**Comment by jezdez** [comment]
@relekang I think it'd be awesome to use sorl-thumbnails' ImageField and admin integration so that we can see the thumbnail there as well. http://sorl-thumbnail.readthedocs.org/en/latest/examples.html

About the 340px width and height, is that related to the 170px max-height currently set to images? 2x for retina? Is this something we should conditionally handle via http://sorl-thumbnail.readthedocs.org/en/latest/template.html#resolution?

**Comment by jezdez** [comment]
Ah, also, is there a reason not to use the thumbnail template tag btw?

**Comment by jezdez** [comment]
Could you set `THUMBNAIL_DEBUG` to `True` in the dev settings?

How about setting `THUMBNAIL_FORMAT` to `'PNG'` so that we can preserve transparency? Alternatively set `THUMBNAIL_PRESERVE_FORMAT` to `True`?

**Comment by relekang** [comment]
> @relekang I think it'd be awesome to use sorl-thumbnails' ImageField and admin integration so that we can see the thumbnail there as well. http://sorl-thumbnail.readthedocs.org/en/latest/examples.html

I can add that :)

>About the 340px width and height, is that related to the 170px max-height currently set to images? 2x for retina? Is this something we should conditionally handle via http://sorl-thumbnail.readthedocs.org/en/latest/template.html#resolution?

I thought that for images that small it would not make a large enough impact to not just use 2x on all instead of adding javascript to check the dpi. I can change that by using `THUMBNAIL_ALTERNATIVE_RESOLUTIONS` and something like [retina.js](http://imulus.github.io/retinajs/).

> Ah, also, is there a reason not to use the thumbnail template tag btw?

Yes, using `[hero.thumbnail for hero in DjangoHero.objects.all().exclude(logo=None)]` in shell will generate all thumbnails, and it makes it easier to test.

> Could you set THUMBNAIL_DEBUG to True in the dev settings?

Sure, I'll fix it later today.

>How about setting THUMBNAIL_FORMAT to 'PNG' so that we can preserve transparency? Alternatively set THUMBNAIL_PRESERVE_FORMAT to True?

Sounds like a good idea.

**Comment by jezdez** [comment]
About checking with retina.js, we're already using Modernizr, so maybe someting like https://github.com/joaocunha/modernizr-retina-test is enough?

**Comment by relekang** [comment]
We still have to change the path of the src on the img tag, but it is just a string replace.

File: fundraising/tests.py [hunk 1]
```diff
@@ -232,6 +234,25 @@ def test_donation_shuffling(self):
                     attrgetter('donated_amount')
                 )
 
+    def test_thumbnail(self):
+        image_path = os.path.join(settings.MEDIA_ROOT, 'fundraising/logos/test_logo.jpg')
```
Comment (jezdez) [inline fundraising/tests.py]: Seems like we need to create that directory first

**Comment by jezdez** [comment]
Nice work @relekang!

File: fundraising/tests.py (hunk 1 above)
Comment (relekang) [inline fundraising/tests.py]: Yeah, somehow I managed [to do this](https://github.com/relekang/djangoproject.com/commit/a387effe289454601c20643812bdf450663f6e75#diff-dd3055d7945a1d7f057a725ff660ef79L238). Fixed it now :)

File: fundraising/tests.py [hunk 2]
```diff
@@ -232,6 +234,29 @@ def test_donation_shuffling(self):
...
+            os.makedirs(os.path.join(settings.MEDIA_ROOT, 'fundraising/logos/'))
+        except OSError:  # directory may already exist
+            pass
+        image_path = os.path.join(settings.MEDIA_ROOT, 'fundraising/logos/test_logo.jpg')
+        image = Image.new('L', (500, 500))
+        image.save(image_path)
+        self.h1.logo = image_path
+        self.h1.save()
+        thumbnail = self.h1.thumbnail
+        self.assertEqual(thumbnail.x, 340)
```
Comment (jezdez) [inline fundraising/tests.py]: Must be 170 :)

Comment (relekang) [inline fundraising/tests.py]: Ah, yeah. Forgot that

**Pull request merged**